garden_ai/entrypoints.py

```diff
@@ -274,6 +274,18 @@ def _is_dlhub_entrypoint(self) -> bool:
         return self.doi in GardenConstants.DLHUB_DOIS
 
 
+class EntrypointIdempotencyError(Exception):
+    """Raised when an entrypoint function is found to be non-idempotent."""
+
+    pass
+
+
+class EntrypointTestError(Exception):
```

Comment:
The `EntrypointIdempotencyError` exception is great but I don't think this one is pulling its weight.

Reply:
Not to belabor the point but in case it's instructive, here's how I'm thinking about it. The main benefits we get from having fine-grained exceptions are:
  - better structure/information in the stacktraces -- it's easier to quickly tell where to start debugging when the bottom of the stacktrace says "EntrypointIdempotencyError" instead of a generic "ValueError" where you have to read up through a few frames of context to see what's happening
  - if we build code on top of this that can tolerate the exception, we want to handle _just_ the specific exception, and let any broader exceptions which we didn't specifically handle propagate up naturally. 
  
So I think this "doesn't pull its weight" in the sense that (a) a user confronted with an `EntrypointTestError` doesn't get more info out of the stacktrace than they would've from the original exception (especially here, since the original came from the user's code) and (b) because it's so general, we can't write more granular exception handling code around it than we could have with a bare Exception

Reply:
for the record I think that the second point doesn't really apply here for either `EntrypointIdempotencyError` or `EntrypointTestError` because they're raised around arbitrary user code, but `EntrypointIdempotencyError` _would_ get us better stacktraces so it's Good

Reply:
That makes sense. Thank you for the feedback!